tests/ttnn/unit_tests/module.mk

```diff
@@ -0,0 +1,29 @@
+
+
+TTNN_UNIT_TEST_DIRS := $(TT_METAL_HOME)/tests/ttnn/unit_tests/ $(TT_METAL_HOME)/tests/ttnn/unit_tests/operations/
+
+TTNN_UNIT_TESTS_SRCS := $(foreach dir,$(TTNN_UNIT_TEST_DIRS),$(wildcard $(dir)*.cpp))
+
+TTNN_UNIT_TESTS_INCLUDES := $(TT_METAL_INCLUDES) $(addprefix -I,$(TTNN_UNIT_TEST_DIRS))
+
+TTNN_UNIT_TESTS_LDFLAGS := -ltt_metal -ldl -lstdc++fs -pthread -lyaml-cpp -lgtest -lm
+
+TTNN_UNIT_TESTS_OBJS := $(addprefix $(OBJDIR)/, $(notdir $(TTNN_UNIT_TESTS_SRCS:.cpp=.o)))
+TTNN_UNIT_TESTS_DEPS := $(TTNN_UNIT_TESTS_OBJS:.o=.d)
+
+-include $(TTNN_UNIT_TESTS_DEPS)
+
+$(OBJDIR)/%.o: %.cpp
+	@mkdir -p $(@D)
+	$(CXX) $(CFLAGS) $(CXXFLAGS) $(TTNN_UNIT_TESTS_INCLUDES) -c -o $@ $<
+
+tests/ttnn: $(TT_METAL_TESTS) tests/ttnn/unit_tests
```

Comment:
You can get rid of `$(TT_METAL_TESTS)` here - you don't depend on anything from those tests, I believe. At least you shouldn't! If there's common stuff, we could uplift that to a common folder.

Also, `tests/ttnn`, `tests/ttnn/all`, `tests/ttnn/%` seem to be generic directives. I would change this to be:

```
# The way we use gtests is to put it all in one exectuable, open to changing this though for ttnn
tests/ttnn/unit_tests: $(TESTDIR)/ttnn/unit_tests

.PRECIOUS: $(TESTDIR)/ttnn/unit_tests
$(TESTDIR)/ttnn/unit_tests: $(TTNN_UNIT_TESTS_OBJS) $(TT_METAL_LIB) $(TT_DNN_LIB) $(TTNN_LIB)
	@mkdir -p $(@D)
	$(CXX) $(CFLAGS) $(CXXFLAGS) -o $@ $^ $(LDFLAGS) $(TTNN_UNIT_TESTS_LDFLAGS)

.PRECIOUS: $(OBJDIR)/ttnn/tests/unit_tests/%.o
$(OBJDIR)/ttnn/tests/unit_tests/%.o: $(OBJDIR)/ttnn/tests/unit_tests/%.cpp
	@mkdir -p $(@D)
	$(CXX) $(CFLAGS) $(CXXFLAGS) $(TTNN_UNIT_TESTS_INCLUDES) -c -o $@ $<
```

Reply:
thank you for your help here.